### PR TITLE
Enhance Cube slicing docs

### PR DESCRIPTION
**Pull request by hsteptoe**
### Description
Some relatively minor enhancements to the docs for the `cube.slices()` and `cube.slices_over()`.

- Made some of the wording clearer
- Modified examples to use `iris-sample-data` 
-  Added example of using dimension indices for slicing.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Comment by hsteptoe** [comment]
**N.B.** I've aimed for judicious use of `# doctest: SKIP` directive to deal with cases where example output would be too long to be useful, or where I seem to get stuck in a linter-modified-test-fail loop with cube printing

File: lib/iris/cube.py [hunk 1]
```diff
@@ -3463,29 +3491,55 @@ def slices(self, ref_to_slice, ordered=True):
...
+            ...     print(sub_cube)  # doctest: SKIP
+
+        To get all realizations as 2x3D separate subcubes, using dimension indices:
+
+            >>> for sub_cube in cube.slices([1, 2, 3]):
+            ...     print(sub_cube)  # doctest: SKIP
 
         See Also
         --------
         iris.cube.Cube.slices :
```
Comment (trexfeathers) [inline lib/iris/cube.py:3539]: Think this should be referencing `slices_over()` instead

File: lib/iris/cube.py [hunk 2]
```diff
@@ -3403,10 +3403,38 @@ def slices_over(self, ref_to_slice):
...
-        For example, to get all subcubes along the time dimension::
+        For example, for a cube with dimensions `realization`, `time`, `latitude` and
+        `longitude`:
 
-            for sub_cube in cube.slices_over('time'):
-                print(sub_cube)
+            >>> fname = iris.sample_data_path('GloSea4', 'ensemble_01[01].pp')
+            >>> cube = iris.load_cube(fname, 'surface_temperature')
+            >>> print(cube)  # doctest: SKIP
+            surface_temperature / (K)           (realization: 2; time: 6; latitude: 145; longitude: 192)
```
Comment (trexfeathers) [inline lib/iris/cube.py:3412]:
```suggestion
        >>> fname = iris.sample_data_path('GloSea4', 'ensemble_01[01].pp')
        >>> cube = iris.load_cube(fname, 'surface_temperature')
        >>> print(cube)
        surface_temperature / (K)           (realization: 2; time: 6; latitude: 145; longitude: 192)
```

File: lib/iris/cube.py [hunk 3]
```diff
@@ -3403,10 +3403,38 @@ def slices_over(self, ref_to_slice):
...
+                0                           time: mean (interval: 1 hour)
+            Attributes:
+                STASH                       m01s00i024
+                source                      'Data from Met Office Unified Model'
+                um_version                  '7.6'
+
+        To get all 12x2D longitude/latitude subcubes:
+
+            >>> for sub_cube in cube.slices_over(['realization', 'time']):
+            ...     print(sub_cube)  # doctest: SKIP
```
Comment (trexfeathers) [inline lib/iris/cube.py:3432]:
```suggestion
        >>> for sub_cube in cube.slices_over(['realization', 'time']):
        ...     print(sub_cube.summary(shorten=True))
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
        surface_temperature / (K)           (latitude: 145; longitude: 192)
```

File: lib/iris/cube.py [hunk 4]
```diff
@@ -3403,10 +3403,38 @@ def slices_over(self, ref_to_slice):
...
+
+        To get realizations as 2x3D separate subcubes, using dimension indices:
+
+            >>> for sub_cube in cube.slices_over(0):
+            ...     print(sub_cube)  # doctest: SKIP
```
Comment (trexfeathers) [inline lib/iris/cube.py:3437]:
```suggestion
        >>> for sub_cube in cube.slices_over(0):
        ...     print(sub_cube.summary(shorten=True))
        surface_temperature / (K)           (time: 6; latitude: 145; longitude: 192)
        surface_temperature / (K)           (time: 6; latitude: 145; longitude: 192)
```

File: lib/iris/cube.py [hunk 5]
```diff
@@ -3463,29 +3491,55 @@ def slices(self, ref_to_slice, ordered=True):
...
-        multi-dimensional cube::
+        For example, for a cube with dimensions `realization`, `time`, `latitude` and
+        `longitude`:
 
-            for sub_cube in cube.slices(['longitude', 'latitude']):
-                print(sub_cube)
+            >>> fname = iris.sample_data_path('GloSea4', 'ensemble_01[01].pp')
+            >>> cube = iris.load_cube(fname, 'surface_temperature')
+            >>> print(cube)  # doctest: SKIP
+            surface_temperature / (K)           (realization: 2; time: 6; latitude: 145; longitude: 192)
```
Comment (trexfeathers) [inline lib/iris/cube.py:3510]:
```suggestion
        >>> fname = iris.sample_data_path('GloSea4', 'ensemble_01[01].pp')
        >>> cube = iris.load_cube(fname, 'surface_temperature')
        >>> print(cube)
        surface_temperature / (K)           (realization: 2; time: 6; latitude: 145; longitude: 192)
```

File: lib/iris/cube.py [hunk 6]
```diff
@@ -3463,29 +3491,55 @@ def slices(self, ref_to_slice, ordered=True):
...
+                0                           time: mean (interval: 1 hour)
+            Attributes:
+                STASH                       m01s00i024
+                source                      'Data from Met Office Unified Model'
+                um_version                  '7.6'
+
+        To get all 12x2D longitude/latitude subcubes:
+
+            >>> for sub_cube in cube.slices(['longitude', 'latitude']):
+            ...     print(sub_cube)  # doctest: SKIP
```
Comment (trexfeathers) [inline lib/iris/cube.py:3530]:
```suggestion
        >>> for sub_cube in cube.slices(['longitude', 'latitude']):
        ...     print(sub_cube.summary(shorten=True))
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
        surface_temperature / (K)           (longitude: 192; latitude: 145)
```

File: lib/iris/cube.py [hunk 7]
```diff
@@ -3463,29 +3491,55 @@ def slices(self, ref_to_slice, ordered=True):
...
+
+        To get all realizations as 2x3D separate subcubes, using dimension indices:
+
+            >>> for sub_cube in cube.slices([1, 2, 3]):
+            ...     print(sub_cube)  # doctest: SKIP
```
Comment (trexfeathers) [inline lib/iris/cube.py:3535]:
```suggestion
        >>> for sub_cube in cube.slices([1, 2, 3]):
        ...     print(sub_cube.summary(shorten=True))
        surface_temperature / (K)           (time: 6; latitude: 145; longitude: 192)
        surface_temperature / (K)           (time: 6; latitude: 145; longitude: 192)
```

File: lib/iris/cube.py [hunk 8]
```diff
@@ -3421,7 +3449,7 @@ def slices_over(self, ref_to_slice):
         iris.cube.Cube.slices :
             Return an iterator of all subcubes given the coordinates or dimension indices.
 
-        """
+        """  # noqa: D214, D406, D407, D410, D411
```
Comment (trexfeathers) [inline lib/iris/cube.py:3452]: This seems to be the current standard to avoid Ruff messing with the `Attributes:` block of a `Cube` summary 👍

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Thanks @hsteptoe, and sorry for the delay.

This is a really valuable addition for an area that I personally regularly forget - examples are gold!

I felt it was important to get the doctests working, to make sure the code continues working in perpetuity. I was able to test and iterate the formatting locally using this handy command (run within an Iris development environment, within the `/docs/src` directory):

```
sphinx-build -b doctest -d _build/doctrees   . _build/doctest generated/api/iris.cube.rst
```

The suggestions work, and pass the pre-commit checks.

File: lib/iris/cube.py (hunk 3 above)
Comment (hsteptoe) [reply lib/iris/cube.py:3432]: Thanks, I wasn't aware of the `shorten=True` keyword!

File: lib/iris/cube.py (hunk 5 above)
Comment (hsteptoe) [reply lib/iris/cube.py]: Done

File: lib/iris/cube.py (hunk 2 above)
Comment (hsteptoe) [reply lib/iris/cube.py]: Done

File: lib/iris/cube.py (hunk 1 above)
Comment (hsteptoe) [reply lib/iris/cube.py:3543]: I think this is OK... could you re-check now the doctest are complete?

File: lib/iris/cube.py [hunk 9]
```diff
@@ -3463,29 +3490,60 @@ def slices(self, ref_to_slice, ordered=True):
...
+        surface_temperature / (K)           (longitude: 192; latitude: 145)
+        surface_temperature / (K)           (longitude: 192; latitude: 145)
+        surface_temperature / (K)           (longitude: 192; latitude: 145)
+        surface_temperature / (K)           (longitude: 192; latitude: 145)
+        surface_temperature / (K)           (longitude: 192; latitude: 145)
+
+
+        .. warning::
+            Note that the dimension order returned in the sub_cubes matches the order specified
+            in the `cube.slices` call, *not* the order of the dimensions in the original cube.
```
Comment (trexfeathers) [inline lib/iris/cube.py:3531]:
```suggestion
            in the ``cube.slices`` call, *not* the order of the dimensions in the original cube.
```

Single backticks just render as italics, so the same as the `*not*` shortly afterwards.

File: lib/iris/cube.py (hunk 1 above)
Comment (trexfeathers) [reply lib/iris/cube.py:3543]: The problem persists, but I think I wasn't clear about which problem I was highlighting!

From `slices_over()`:

https://github.com/SciTools/iris/blob/f06aa462d9bd7caa2c3513689b3d8a1e8d5a8fba/lib/iris/cube.py#L3446-L3449

From `slices()`:

https://github.com/SciTools/iris/blob/f06aa462d9bd7caa2c3513689b3d8a1e8d5a8fba/lib/iris/cube.py#L3541-L3544

I believe the documentation for `slices()` should be referencing `slices_over()`, not itself?

Also totally not your fault, but while we're here...

File: lib/iris/cube.py (hunk 9 above)
Comment (trexfeathers) [inline lib/iris/cube.py:3531]:
```suggestion
            in the ``cube.slices`` call, *not* the order of the dimensions in the original cube.
```

Single backticks just render as italics, so the same as the `*not*` shortly afterwards.

File: lib/iris/cube.py (hunk 1 above)
Comment (hsteptoe) [reply lib/iris/cube.py:3543]: Ah yes, I see. Fixed

File: lib/iris/cube.py (hunk 9 above)
Comment (hsteptoe) [reply lib/iris/cube.py]: Fixed

**Review by trexfeathers — APPROVED** [review]

**Comment by trexfeathers** [comment]
Brilliant, thanks again!

**Pull request merged**